fdbcli/MetaclusterCommands.actor.cpp

```diff
@@ -287,9 +287,47 @@ ACTOR Future<bool> metaclusterStatusCommand(Reference<IDatabase> db, std::vector
 
 	state bool useJson = tokens.size() == 3;
 
+	state Reference<ITransaction> tr = db->createTransaction();
+	tr->setOption(FDBTransactionOptions::READ_SYSTEM_KEYS);
+	tr->setOption(FDBTransactionOptions::RAW_ACCESS);
+	state Future<Optional<MetaclusterRegistrationEntry>> metaclusterRegistrationFuture =
+	    MetaclusterAPI::getMetaclusterRegistrationEntryTransaction(tr);
+	Optional<MetaclusterRegistrationEntry> metaclusterRegistrationEntry = wait(metaclusterRegistrationFuture);
+
+	const ClusterType clusterType = !metaclusterRegistrationEntry.present()
+	                                    ? ClusterType::STANDALONE
+	                                    : metaclusterRegistrationEntry.get().clusterType;
+	if (ClusterType::STANDALONE == clusterType) {
+		if (useJson) {
+			json_spirit::mObject obj;
+			obj["type"] = "success";
+			obj["management_cluster"] = "null";
+			fmt::print("{}\n", json_spirit::write_string(json_spirit::mValue(obj), json_spirit::pretty_print).c_str());
+		} else {
+			fmt::print("A standalone cluster, not part of any metacluster\n");
+		}
+		wait(safeThreadFutureToFuture(tr->commit()));
+		return true;
+	} else if (ClusterType::METACLUSTER_DATA == clusterType) {
+		if (useJson) {
+			json_spirit::mObject obj;
+			obj["type"] = "success";
+			obj["management_cluster"] = metaclusterRegistrationEntry.get().metaclusterName.toString();
+			fmt::print("{}\n", json_spirit::write_string(json_spirit::mValue(obj), json_spirit::pretty_print).c_str());
+		} else {
+			fmt::print(
+			    "This cluster \"{}\" is a data cluster within a metacluster whose management cluster is \"{}\"\n",
+			    metaclusterRegistrationEntry.get().name.toString().c_str(),
+			    metaclusterRegistrationEntry.get().metaclusterName.toString().c_str());
+		}
+		wait(safeThreadFutureToFuture(tr->commit()));
+		return true;
+	} else {
+		ASSERT(ClusterType::METACLUSTER_MANAGEMENT == clusterType);
+	}
 	try {
```

Comment:
If we're extending the scope of the transaction, we should include all of the transaction operations in the try block so that we can handle expected errors.

Reply:
Done.